wagtailvideos/models.py

```diff
@@ -256,9 +256,23 @@ def __init__(self, transcode, **kwargs):
         self.transcode = transcode
 
     def run(self):
+        # Check if Django Storages is installed and if S3 is available.
+        S3Boto3Storage = True
+        try:
+            from storages.backends.s3boto3 import S3Boto3Storage
+        except ImportError:
+            S3Boto3Storage = False
+
         video = self.transcode.video
         media_format = self.transcode.media_format
-        input_file = video.file.path
+
+        # If the file is stored on S3, we need to use the URL instead of the path.
+        storage_backend = video.file.storage
+        if S3Boto3Storage and isinstance(storage_backend, S3Boto3Storage):
```

Comment:
We can probably condense this down a bit to remove the import check above -

```python
try:
    input_file = video.file.path
except NotImplementedError:
    input_file = vide.file.url 
```

I had no idea ffmpeg works with a url as well as a file path. I'm assuming this all works ok? I don't really have an easy way to test it

Reply:
Maybe, the condensed version is what I had in the beginning, but I am not sure that `NotImplementedError` is always due to the absolute path exception. In other words, I feel this is a bit too broad and does not guarantee we have `video.file.url` available. That is why I check if Storages is installed and specifically check on the S3 backend being set so I know for sure that we have a `url` attribute on `video.file`.

And yes, this is tested, ffmpeg accepts URLs just fine as input.

Reply:
I'm ok with the broad approach, it's basically the same check that Wagtail does - https://github.com/wagtail/wagtail/blob/main/wagtail/images/models.py#L133